Bottom: c9fd427003033aae95052e015dcb56af70d55007
Top:    f295b2654243a5471ca12f9f8ae835a5513df185
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2012-11-21 21:21:36 +0000

TMP FIXES


---

diff --git a/src/draw_funcs.c b/src/draw_funcs.c
index 9edc038..e4a1d86 100644
--- a/src/draw_funcs.c
+++ b/src/draw_funcs.c
@@ -432,8 +432,12 @@ draw_layer (LayerType *layer, const BoxType *drawn_area, void *userdata)
   int layer_num = GetLayerNumber (PCB->Data, layer);
   int group = GetLayerGroupNumberByPointer (layer);
   struct poly_info info = {drawn_area, layer};
+  bool is_outline;
 
-  if (layer_num < max_copper_layer)
+  is_outline = strcmp (layer->Name, "outline") == 0 ||
+               strcmp (layer->Name, "route") == 0;
+
+  if (layer_num < max_copper_layer && !is_outline)
     {
       r_search (PCB->Data->pin_tree, drawn_area, NULL, pin_hole_callback, NULL);
       r_search (PCB->Data->via_tree, drawn_area, NULL, via_hole_callback, NULL);
@@ -445,12 +449,6 @@ draw_layer (LayerType *layer, const BoxType *drawn_area, void *userdata)
   if (TEST_FLAG (CHECKPLANESFLAG, PCB))
     return;
 
-  if (layer_num < max_copper_layer)
-    {
-      r_search (PCB->Data->pin_tree, drawn_area, NULL, pin_hole_callback, NULL);
-      r_search (PCB->Data->via_tree, drawn_area, NULL, via_hole_callback, NULL);
-    }
-
   /* draw all visible lines this layer */
   r_search (layer->line_tree, drawn_area, NULL, line_callback, layer);
   r_search (layer->arc_tree,  drawn_area, NULL, arc_callback,  layer);
@@ -460,8 +458,7 @@ draw_layer (LayerType *layer, const BoxType *drawn_area, void *userdata)
      auto-outline magically disappear when you first add something to
      the "outline" layer.  */
 
-  if (strcmp (layer->Name, "outline") == 0 ||
-      strcmp (layer->Name, "route") == 0)
+  if (is_outline)
     {
       if (IsLayerEmpty (layer))
         {
